tests/test_misc.py

```diff
@@ -1439,3 +1439,104 @@ def test_large_spends(revault_network, bitcoind, executor):
+    """
+    revault_network.deploy(3, 1)
+    man = revault_network.man_wallets[0]
+    stks = revault_network.stk_wallets
+    vault = revault_network.fund(18)
+    deposit = f"{vault['txid']}:{vault['vout']}"
+    stks[0].wait_for_deposits([deposit])
+
+    # Can't cancel an unconfirmed deposit
+    with pytest.raises(RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"):
+        stks[0].rpc.revault(deposit)
+
+    # A manager gets the same error: both parties can revault
+    with pytest.raises(RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"):
+        man.rpc.revault(deposit)
+
+    child_index = vault["derivation_index"]
+    psbts = stks[0].rpc.getrevocationtxs(deposit)
+    cancel_psbt = psbts["cancel_tx"]
+    emer_psbt = psbts["emergency_tx"]
+    unemer_psbt = psbts["emergency_unvault_tx"]
+    for stk in stks:
+        cancel_psbt = stk.stk_keychain.sign_revocation_psbt(
+            cancel_psbt, child_index
+        )
+        emer_psbt = stk.stk_keychain.sign_revocation_psbt(emer_psbt, child_index)
+        unemer_psbt = stk.stk_keychain.sign_revocation_psbt(
+            unemer_psbt, child_index
+        )
+    stks[0].rpc.revocationtxs(deposit, cancel_psbt, emer_psbt, unemer_psbt)
+    assert stks[0].rpc.listvaults([], [deposit])["vaults"][0]["status"] == "secured"
+
+    # Secured is not good enough though
+    with pytest.raises(RpcError, match="Invalid vault status: 'secured'. Need 'unvaulting'"):
+        stks[0].rpc.revault(deposit)
+
+    unvault_psbt = stks[0].rpc.getunvaulttx(deposit)["unvault_tx"]
+
+    for stk in stks:
+        wait_for(
+            lambda: stk.rpc.listvaults([], [deposit])["vaults"][0]["status"]
+            == "secured"
+        )
+        unvault_psbt = stk.rpc.getunvaulttx(deposit)["unvault_tx"]
+        unvault_psbt = stk.stk_keychain.sign_unvault_psbt(unvault_psbt, child_index)
+        stk.rpc.unvaulttx(deposit, unvault_psbt)
+
+    for stk in stks:
+        wait_for(
+            lambda: stk.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "active"
+        )
+
+    # Active? Not enough!
+    with pytest.raises(RpcError, match="Invalid vault status: 'active'. Need 'unvaulting'"):
+        stks[0].rpc.revault(deposit)
+
+    # Now we want to broadcast the unvault tx without having an associated spend tx
+    # First of all, we need the unvault psbt finalized
+    unvault_psbt = stks[0].rpc.listpresignedtransactions([deposit])['presigned_transactions'][0]['unvault']
+    unvault_tx = bitcoind.rpc.finalizepsbt(unvault_psbt)['hex']
+    bitcoind.rpc.sendrawtransaction(unvault_tx)
+
+    # Unvaulting! And there's no associated spend tx! Is revault broken?
+    for stk in stks:
+        wait_for(
+            lambda: stk.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "unvaulting"
+        )
+
+    # Nah it's not, just broadcast the cancel
+    stks[0].rpc.revault(deposit)
+
+    time.sleep(5)
```

Comment:
Don't sleep, add debug logs in revaultd and use `wait_for_logs`. But anyways i think you can strip the remaining part with a `TODO: Cancel state tracking` for this commit and add proper testing when you implement it afterwards